src/olympia/addons/indexers.py

```diff
@@ -388,3 +393,120 @@ def extract_document(cls, obj):
+                    'unique'
+                ]
+            }
+        },
+        'normalizer': {
+            'lowercase_keyword_normalizer': {
+                'type': 'custom',
+                'filter': ['lowercase'],
+            },
+        },
+        'filter': {
+            'wordDelim': {
+                # This filter is useful for add-on names that have multiple
+                # words sticked together in a way that is easy to recognize,
+                # like FooBar, which should be indexed as FooBar and Foo Bar.
+                # (preserve_original: True makes us index both the original
+                # and the split version.)
+                'type': 'word_delimiter',
+                'preserve_original': True
+            },
+            'dict': {
+                # This filter is also useful for add-on names that have
+                # multiple words sticked together, but without a pattern that
+                # we can automatically recognize. To deal with those, we use
+                # a small dictionary of common words. It allows us to index
+                # "awesometabpassword"  as "awesome tab password", helping
+                # users looking for "tab password" find that add-on.
+                'type': 'dictionary_decompounder',
+                'word_list': [
+                    'all', 'auto', 'ball', 'bar', 'block', 'blog', 'bookmark',
+                    'browser', 'bug', 'button', 'cat', 'chat', 'click', 'clip',
+                    'close', 'color', 'context', 'cookie', 'cool', 'css',
+                    'delete', 'dictionary', 'down', 'download', 'easy', 'edit',
+                    'fill', 'fire', 'firefox', 'fix', 'flag', 'flash', 'fly',
+                    'forecast', 'fox', 'foxy', 'google', 'grab', 'grease',
+                    'html', 'http', 'image', 'input', 'inspect', 'inspector',
+                    'iris', 'js', 'key', 'keys', 'lang', 'link', 'mail',
+                    'manager', 'map', 'mega', 'menu', 'menus', 'monkey',
+                    'name', 'net', 'new', 'open', 'password', 'persona',
+                    'privacy', 'query', 'screen', 'scroll', 'search', 'secure',
+                    'select', 'smart', 'so', 'spring', 'status', 'style',
+                    'super', 'sync', 'tab', 'text', 'think', 'this', 'time',
+                    'title', 'translate', 'tree', 'undo', 'up', 'upload',
+                    'url', 'user', 'video', 'window', 'with', 'word', 'zilla',
+                ]
+            }
+        }
+    }
+}
+
+
+def create_new_index(index_name=None):
+    """
+    Create a new index for addons in ES.
+
+    Intended to be used by reindexation (and tests), generally a bad idea to
+    call manually.
+    """
+    if index_name is None:
+        index_name = AddonIndexer.get_index_alias()
+
+    index_settings = copy.deepcopy(INDEX_SETTINGS)
+
+    if waffle.switch_is_active('es-use-classic-similarity'):
```

Comment:
Do we need this waffle flag?  classic is either better or not, I'd guess.

Reply:
The waffle is currently on in prod. It has helped our relevancy problems a bit, but eventually it will need to be removed as it's deprecated in future versions of ES.

Evaluating whether or not we can remove it soon is on my TODO list - We'll see once I've done all the work I want to do for this quarter.

(In any case, I'm just moving the contents of another file here, it's not new code)